### PR TITLE
add_support_for_Samsung_SSD_and_refactoring

### PR DESCRIPTION
**Pull request by zakinko**

### CODE REVIEW

**Comment by zakinko** [comment]
Tested with many Samsung SSD, please ask if you need anything else.

**Comment by samm-git** [comment]
Thank you for the contribution. Please ignore failed build, it is expected. I will review it and merge to the SVN.

**Comment by samm-git** [comment]
@zakinko please provide a feedback when possible

File: smartmontools/drivedb.h [hunk 1]
```diff
@@ -1378,48 +1378,150 @@ const drive_settings builtin_knowndrives[] = {
     "-v 242,raw48,Host_Reads_32MiB"
   },
   { "Samsung based SSDs",
-    "SAMSUNG SSD PM800 .*GB|"  // SAMSUNG PM800 SSDs, tested with SAMSUNG SSD PM800 TH 64GB/VBM25D1Q
```
[the remark this answers is not in the document]
Comment (vinibali) [reply smartmontools/drivedb.h:1381]: What happened with this pull request?
I have an old MMCRE28G5MXP, which doesn't exist in the latest drivedb.

**Pull request closed without merging**